packages/editor/src/components/rich-text/index.js

```diff
@@ -738,13 +743,66 @@ export class RichText extends Component {
 		}
 	}
 
+	/**
+	 * Applies given annotations to the given record.
+	 *
+	 * @param {Object} record The record to apply annotations to.
+	 * @param {Array} annotations The annotation to apply.
+	 * @return {Object} A record with the annotations applied.
+	 */
+	applyAnnotations( record, annotations ) {
+		annotations
+			.map( ( annotation ) => {
+				let startPos = matchXPath( record, annotation.startXPath );
+				let endPos = matchXPath( record, annotation.endXPath );
+
+				if ( startPos !== false ) {
+					startPos += annotation.startOffset;
+				}
+
+				if ( endPos !== false ) {
+					endPos += annotation.endOffset;
+				}
+
+				if (
+					startPos !== false &&
+					endPos !== false &&
+					startPos <= endPos &&
+					startPos <= record.text.length &&
+					endPos <= record.text.length
+				) {
+					return {
+						start: startPos,
+						end: endPos,
+						className: 'annotation-text annotation-text-' + annotation.source,
+					};
+				}
+
+				return false;
+			} )
+			.filter( Boolean )
+			.forEach( ( { start, end, className } ) => {
+				record = applyFormat(
+					record,
+					{ type: 'core/annotation', attributes: { className } },
+					start,
+					end
+				);
+			} );
+
+		return record;
+	}
+
 	componentDidUpdate( prevProps ) {
-		const { tagName, value, isSelected } = this.props;
+		const { tagName, value, isSelected, annotations } = this.props;
+
+		const applyAnnotations = annotations !== prevProps.annotations;
```

Comment:
This name is confusing as it is also the name of the class method. `shouldApplyAnnotations` can be an alternative or it can be inlined if it isn't used in other places.